src/components/search_bar/search_bar.tsx

```diff
@@ -40,25 +43,33 @@ class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
         }
     }
 
+    componentDidUpdate(prevProps: SearchBarProps, prevState: SearchBarState) {
+        const { query } = this.props;
+        if (query !== prevProps.query && query !== prevState.query) {
+            // eslint-disable-next-line react/no-did-update-set-state
+            this.setState({
+                query,
+            });
+        }
+    }
+
     componentWillUnmount() {
         this.debouncedEmit.cancel();
     }
 
     onChange(event: React.ChangeEvent<HTMLInputElement>) {
         const { value: query } = event.target;
-
         this.setState({
             query,
+        }, () => {
+            this.debouncedEmit(query);
```

Comment:
Ha!

Reply:
In practice this made no difference but it's still nicer to make sure that the debounced handler is called after state was updated.